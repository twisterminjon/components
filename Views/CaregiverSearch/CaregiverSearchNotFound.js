import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Button } from 'semantic-ui-react'

import './CaregiverSearch.css'

export default class CaregiverSearchNotFound extends Component {
  static propTypes = {
    /** Called after the add button is clicked */
    onAdd: PropTypes.func.isRequired,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
  }
  static defaultProps = {
    className: '',
    style: {},
  }

  render() {
    const { onAdd, className, style } = this.props
    return (
      <div className={`caregiversearchnotfound ${className}`.trim()} style={style}>
        <p>We couldnâ€™t find a Caregiver with that phone number.</p>
        <Button inverted className="brand-button" onClick={onAdd} data-testid="add-caregiver">
          Create a Caregiver with that number?
        </Button>
      </div>
    )
  }
}
