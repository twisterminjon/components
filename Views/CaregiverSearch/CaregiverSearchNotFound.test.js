import React from 'react'

import CaregiverSearchNotFound from './CaregiverSearchNotFound'

describe('CaregiverSearchNotFound', () => {
  const mockAdd = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<CaregiverSearchNotFound onAdd={mockAdd} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('clicking the add button calls the onAdd function', () => {
    const wrapper = window.mount(<CaregiverSearchNotFound onAdd={mockAdd} />)

    wrapper
      .find('[data-testid="add-caregiver"]')
      .at(1)
      .simulate('click')

    expect(mockAdd).toHaveBeenCalled()
  })
})
