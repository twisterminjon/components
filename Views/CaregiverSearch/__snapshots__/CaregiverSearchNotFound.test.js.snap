// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaregiverSearchNotFound matches the snapshot 1`] = `
<div
  className="caregiversearchnotfound"
  style={Object {}}
>
  <p>
    We couldnâ€™t find a Caregiver with that phone number.
  </p>
  <Button
    as="button"
    className="brand-button"
    data-testid="add-caregiver"
    inverted={true}
    onClick={[MockFunction]}
  >
    Create a Caregiver with that number?
  </Button>
</div>
`;
